src/v/compression/async_stream_zstd.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+#include "bytes/iobuf.h"
+#include "utils/mutex.h"
+
+#include <seastar/core/aligned_buffer.hh>
+
+#include <memory>
+#include <zstd.h>
+
+namespace compression {
+class async_stream_zstd {
+public:
+    ss::future<iobuf> compress(const iobuf& b) { return do_compress(b); }
+    ss::future<iobuf> uncompress(const iobuf& b) { return do_uncompress(b); }
+    ss::future<iobuf> compress(iobuf&& b) { return do_compress(b); }
+    ss::future<iobuf> uncompress(iobuf&& b) { return do_uncompress(b); }
```

Comment:
take `b` by value. iobuf is already move-only

Reply:
Switching over `async_stream_zstd`'s API to take `b` by value only.